src/firmware/boards/legacy_robot_stm32f4/io/receive.c

```diff
@@ -105,13 +108,19 @@ static void receive_task(void *UNUSED(param))
                 {
                     // Broadcast frame must contain a camera packet or drive packet
                     // Note that camera packets have a variable length.
-                    if (dma_buffer[MESSAGE_PURPOSE_ADDR] == 0x0FU)
+                    if (dma_buffer[MESSAGE_PURPOSE_INDEX] == 0x0FU)
                     {
-                        handle_drive_packet(dma_buffer);
+                        iprintf("Frame length: %d\r\n", frame_length);
+                        iprintf("Frame header length: %d\r\n", HEADER_LENGTH);
+                        iprintf("Frame footer length: %d\r\n", FOOTER_LENGTH);
+                        iprintf("MESSAGE_PURPOSE_INDEX: %d\r\n", MESSAGE_PURPOSE_INDEX);
+                        handle_drive_packet(
```

Comment:
Assuming this is debug code that will be deleted?

Reply:
lol it is now, thanks, removed